### ISSUE TITLE
Leading single quote in constant expression is parsed silently

### ISSUE THREAD

**Issue by amiralies** [rescript-lang/syntax#163]
```rescript
let a = '"foo"
let b = 'true
let c = '2
let d = '2.2
let e = '''
````
is parsed into:
```rescript
let a = "foo"
let b = true
let c = 2
let d = 2.2
let e = '\''

```

**Comment by amiralies** [comment]
I saw that there's a `SingleQuote` token in scanner. can we remove that and parse to more complex tokens directly?

**Comment by IwanKaramazow** [comment]
Thanks for the report, will see what's best.

**Comment by amiralies** [comment]
looks like it's a bit more general than what explained above.
for example this snippet
```rescript
let a =    =  =  = =  = =   '  ' }  } } } ] | | & * )   } =  = = =(x==y)
```
is being parsed into
```rescript
let a = x == y
```

**Comment by amiralies** [comment]
Fixed via #279
